gitpod-io/gitpod
# [code] github auth provider

#### What it does

This PR hooks our generic auth provider as github auth provider in VS Code. The default github authentication extension is excluded. `github.vscode-pull-request-github` is installed if a workspace is started from github.com context. You can actually review this PR using this PR from VS Code: https://ak-github-auth.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod/pull/3349

Additionally it fixes bugs in the git token provider of supervisor which before will return any token regardless of given scopes.

Changes in code: https://github.com/gitpod-io/vscode/compare/gp-code...ak/generic_auth

#### How to test

- Switch to Code
- Start a GitHub workspace
- Check that GitHub pull request review extension is installed but not signed in
- Check that pushing from the cli is not allowed, not enough permissions
- Go to Access Control and grant access to public and private repos
- Sign in the pull request review extension or reload the page, check that it works now
- Check that pushing from the cli now works
- Reload the page check again
- Restart the workspace check again
- Start another workspace and check again, like on this PR to check out PR review mode: https://ak-github-auth.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod/pull/3349
